integration-tests/mpp/android-module/build.gradle.kts

```diff
@@ -0,0 +1,88 @@
+import com.android.builder.core.BuilderConstants
+import com.squareup.anvil.plugin.AndroidVariantFilter
+
+plugins {
+  alias(libs.plugins.kotlin.multiplatform)
+  alias(libs.plugins.kotlin.kapt)
+  alias(libs.plugins.agp.library)
+  id("com.squareup.anvil")
+  id("conventions.minimal")
+}
+
+android {
+  compileSdk = 33
+  namespace = "com.squareup.anvil.mpp"
+
+  defaultConfig {
+    minSdk = 24
+    @Suppress("UnstableApiUsage")
+    targetSdk = 33
+  }
+
+  compileOptions {
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
+  }
+
+  lint {
+    @Suppress("UnstableApiUsage")
+    warningsAsErrors = true
+  }
+}
+
+androidComponents {
+
+  beforeVariants { variant ->
+    variant.enable = variant.buildType != BuilderConstants.RELEASE
+  }
+}
+
+kotlin {
+
+  // The `android()` target is deprecated in Kotlin 1.9.0+,
+  // but the new `androidTarget()` alternative doesn't exist in KGP 1.8.x.
+  @Suppress("DEPRECATION")
+  android()
+
+  sourceSets {
+    val androidMain by getting {
+      dependencies {
+
+        val kapt by configurations.getting
+        kapt.dependencies.addLater(libs.dagger2.compiler)
```

Comment:
Why do we apply this here compared to before when it was in the `dependencies` block at the end of this file? And why `addLater` instead of directly applying?

Reply:
> Why do we apply this here compared to before when it was in the dependencies block at the end of this file?

I had been experimenting with a few different ways to make it work without the weird syntax, though they all failed.  I'll move it back.

> And why addLater instead of directly applying?

This is the standard behavior, but normally it's hidden by many layers of `dependencyNotation: Any` parameters.

All the dependencies from `libs` are some variation of `Provider<Dependency>`.  At this layer, the normal `dependencies.add(...)` only accepts a `Dependency` argument, and all those providers are added via `addLater(...)`.